app/renderer/js/components/context-menu.ts

```diff
@@ -0,0 +1,105 @@
+import {remote} from 'electron';
+import WebView from './webview';
+import * as t from '../utils/translation-util';
+const {clipboard, Menu} = remote;
+
+export const contextMenu = (webview: WebView, event: any) => {
```

Comment:
Avoid `any` if at all possible. `event` has type `Event`. There is no `params` member of `Event`, but the `context-menu` event handler is passed `params: ContextMenuParams` as a second parameter. (Reference: https://github.com/electron-userland/electron-builder/blob/master/typings/electron.d.ts.)

This also makes the below `selectionText as string` casts unnecessary.

Reply:
@andersk Sorry but I'm having a little hard time getting to this. Can you please give some suggestion?

* The listener here is from `WebContents` class. Though we can definitely pass the webContents of the webview as that's what we're operating on, but the listener won't call our context-menu component.
* How is it obtaining the `params` object which is being passed as `ContextMenuParams`?
* Also keeping in mind the ContextMenuParams has everything we need but the `dictionarySuggestions[]`